frontend/public/module/k8s/index.ts

```diff
@@ -54,7 +55,7 @@ export type K8sResourceCondition<T> = {
   message: string;
 };
 
-export type MatchExpression = {key: string, operator: 'Exists' | 'DoesNotExist'} | {key: string, operator: 'In' | 'NotIn' | 'Equals' | 'NotEquals', values: string[]};
+export type MatchExpression = {key: string, operator: 'Exists' | 'DoesNotExist' | 'In' | 'NotIn' | 'Equals' | 'NotEquals' | string, values?: string[]};
```

Comment:
to conform to the API: 
please see `"io.k8s.apimachinery.pkg.apis.meta.v1.LabelSelectorRequirement": {` in https://raw.githubusercontent.com/openshift/origin/master/api/swagger-spec/openshift-openapi-spec.json

Reply:
The type for `MatchExpression` was taken from the [Kubernetes label selector docs](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements), which indicate that if you are using `Exists`/`DoesNotExist`, than `values` is not required. For all other operators, `values` is required. What is proposed here is less correct.

Reply:
I found out that the original TC does not work for PVCs (https://kubernetes.io/docs/concepts/storage/persistent-volumes/#selector). So that is why I made a looser version of this type. 

Please let me know if I should choose some other definition which would conform to all of these.

Reply:
Can you give me an example of where this type doesn't work? I know the kube API has some "inconsistencies", but I'm pretty sure this type is correct.

Reply:
When trying to use the generated types (please see https://github.com/openshift/console/pull/1813/files#diff-bc38bd73768fb0fddffa9019eab829e1) as it clashes with our `K8sResourceKind` in function parameters (e.g. `getName` in `@console/shared`).

Should I remove this change from this PR and wait for #1813 to get in first?

Reply:
Ugh, the OpenAPI definitions are _less_ strict because they put the acceptable operators in the description instead of making it a concrete type, like we have here. 
Do you mind removing the change here and we'll deal with it in #1813?

Reply:
Ok, removed. Let's resolve this later.